### ISSUE TITLE
frawk does not detect start of program without -- when -v is last argument before program start.

### ISSUE THREAD

**Issue by ghuls** [ezrosent/frawk#22]
frawk does not detect start of program without -- when -v is last argument before program start.

```
❯ seq 1 10 | frawk -v multiplier=20 '{ print $0 * multiplier }'
must specify program at command line, or in a file via -f

❯ seq 1 10 | frawk -v multiplier=20 -- '{ print $0 * multiplier }'
20.0
40.0
60.0
80.0
100.0
120.0
140.0
160.0
180.0
200.0

❯ seq 1 10 | frawk -F '\t' -v multiplier=20 '{ print $0 * multiplier }'
must specify program at command line, or in a file via -f

~/Downloads via ☕ v1.8.0 via 🐍 v3.6.7
❯ seq 1 10 | frawk -v multiplier=20 -F '\t' '{ print $0 * multiplier }'
20.0
40.0
60.0
80.0
100.0
120.0
140.0
160.0
180.0
200.0
```

frawk also seems to have problems with parsing variables with dashes (and maybe other special characters):
```
❯ seq 1 10 | awk -v multiplier=20 -v var_with_dash=variable-with-dash -F '\t' '{ print $0 * multiplier, var_with_dash }'
20 variable-with-dash
40 variable-with-dash
60 variable-with-dash
80 variable-with-dash
100 variable-with-dash
120 variable-with-dash
140 variable-with-dash
160 variable-with-dash
180 variable-with-dash
200 variable-with-dash

❯ seq 1 10 | frawk -v multiplier=20 -v var_with_dash=variable-with-dash -F '\t' '{ print $0 * multiplier, var_with_dash }'
failed to parse var at index 2:
var_with_dash=variable-with-dash
error:UnrecognizedToken { token: (Loc { line: 0, col: 22, offset: 22 }, Sub, Loc { line: 0, col: 23, offset: 23 }), expected: ["\"[\""] }

❯ seq 1 10 | frawk -v multiplier=20 -v 'var_with_dash=variable-with-dash' -F '\t' '{ print $0 * multiplier, var_with_dash }'
failed to parse var at index 2:
var_with_dash=variable-with-dash
error:UnrecognizedToken { token: (Loc { line: 0, col: 22, offset: 22 }, Sub, Loc { line: 0, col: 23, offset: 23 }), expected: ["\"[\""] }

❯ seq 1 10 | frawk -v multiplier=20 -v 'var_with_dash="variable-with-dash"' -F '\t' '{ print $0 * multiplier, var_with_dash }'
20.0 variable-with-dash
40.0 variable-with-dash
60.0 variable-with-dash
80.0 variable-with-dash
100.0 variable-with-dash
120.0 variable-with-dash
140.0 variable-with-dash
160.0 variable-with-dash
180.0 variable-with-dash
200.0 variable-with-dash

❯ seq 1 10 | frawk -v multiplier=20 -v var_with_dash="variable-with-dash" -F '\t' '{ print $0 * multiplier, var_with_dash }'
failed to parse var at index 2:
var_with_dash=variable-with-dash
error:UnrecognizedToken { token: (Loc { line: 0, col: 22, offset: 22 }, Sub, Loc { line: 0, col: 23, offset: 23 }), expected: ["\"[\""] }
```

If passed vars are not a number, the variable will be empty if not quoted (once for bash and once for frawk)). Which is quite different from normal awk.
```
❯ seq 1 10 | frawk -v multiplier=20 -v X=A -v 'var_with_dash="variable-with-dash"' -F '\t' '{ print $0 * multiplier, X, var_with_dash }'
20.0  variable-with-dash
40.0  variable-with-dash
60.0  variable-with-dash
80.0  variable-with-dash
100.0  variable-with-dash
120.0  variable-with-dash
140.0  variable-with-dash
160.0  variable-with-dash
180.0  variable-with-dash
200.0  variable-with-dash

❯ seq 1 10 | frawk -v multiplier=20 -v 'X=A' -v 'var_with_dash="variable-with-dash"' -F '\t' '{ print $0 * multiplier, X, var_with_dash }'
20.0  variable-with-dash
40.0  variable-with-dash
60.0  variable-with-dash
80.0  variable-with-dash
100.0  variable-with-dash
120.0  variable-with-dash
140.0  variable-with-dash
160.0  variable-with-dash
180.0  variable-with-dash
200.0  variable-with-dash

❯ seq 1 10 | frawk -v multiplier=20 -v 'X="A"' -v 'var_with_dash="variable-with-dash"' -F '\t' '{ print $0 * multiplier, X, var_with_dash }'
20.0 A variable-with-dash
40.0 A variable-with-dash
60.0 A variable-with-dash
80.0 A variable-with-dash
100.0 A variable-with-dash
120.0 A variable-with-dash
140.0 A variable-with-dash
160.0 A variable-with-dash
180.0 A variable-with-dash
200.0 A variable-with-dash
```

**Comment by ezrosent** [comment]
Thanks again! I had never used the behavior you've shown with "variable-with-dash;" I'll try my best to add support for this.

As for the `--` interaction, I'm hoping this isn't a limitation of the library frawk uses for command-line parsing. I'll look into what it would take to allow a program to come after `-v` without an intervening `--`.

**Comment by ghuls** [comment]
In my script I had 2 variables with a dash as I they both contained a filename to which data will be written based on conditions in the main awk body.

**Comment by ghuls** [comment]
There is also another problem with dashes when dusing the `-O` option with `-1` as it wants to parse it as an option too, if it is separated by a space.

```
❯ printf '1234567890\n' |  target/release/frawk -O -1 '{ print substr($0, 3, 2)}'
error: Found argument '-1' which wasn't expected, or isn't valid in this context

If you tried to supply `-1` as a PATTERN use `-- -1`

USAGE:
    frawk [FLAGS] [OPTIONS] [--] [ARGS]

For more information try --help

❯ printf '1234567890\n' |  target/release/frawk -O-1 '{ print substr($0, 3, 2)}'
34

❯ printf '1234567890\n' |  target/release/frawk -O2 '{ print substr($0, 3, 2)}'
34

❯ printf '1234567890\n' |  target/release/frawk -O 2 '{ print substr($0, 3, 2)}'
34

❯ printf '1234567890\n' |  target/release/frawk -O-1 '{ print substr($0, 3)}'
34567890
```

**Comment by ghuls** [comment]
`-v` seems to be completely broken at the moment:

```
❯ seq 1 10 | frawk -v multiplier=20 -v 'X="A"' -v 'var_with_dash="variable-with-dash"' -F '\t' '{ print $0 * multiplier, X, var_with_dash }'
failed to parse var at index 1:
multiplier=20
error:UnrecognizedToken { token: (Loc { line: 0, col: 13, offset: 13 }, Newline, Loc { line: 0, col: 13, offset: 13 }), expected: ["\"[\""] }
```

**Comment by ezrosent** [comment]
Thanks for the test case! sorry for the breakage. This is a hole in frawk's test coverage, but we now have end to end tests set up so once this is fixed (as well as the original bug) it won't regress.

**Comment by ghuls** [comment]
`--var VAR=VAL` option does not exist in other awk versions as far as I know. `gawk` uses `--assign VAR=VAL` as long option, so I think it would be better to mimic that behavior if you add a long option.

**Comment by ezrosent** [comment]
Thanks. I went ahead and removed the long option. I think this issue should be fixed as of #39, so I'm closing the issue. Of course, feel free to leave a comment/open a new issue if it looks like I missed something.

**Comment by ghuls** [comment]
This is now fixed in: https://github.com/ezrosent/frawk/pull/81
